framework/queue/Queue.php

```diff
@@ -0,0 +1,47 @@
+<?php
+/**
+ * @link http://www.yiiframework.com/
+ * @copyright Copyright (c) 2008 Yii Software LLC
+ * @license http://www.yiiframework.com/license/
+ */
+
+namespace yii\queue;
+
+use yii\base\Component;
+
+/**
+ * Queue is the base class for queue classes supporting different queue back-ends.
+ *
+ * @author Alexander Kochetov <creocoder@gmail.com>
+ * @since 2.0.5
+ */
+abstract class Queue extends Component
+{
+    /**
+     * Push a new job to the queue.
+     *
+     * @param string $job
+     * @param mixed|null $data
+     * @param string|null $queue
+     * @param integer $delay
+     */
+    abstract public function push($job, $data = null, $queue = null, $delay = 0);
```

Comment:
Because `Yii::$app->queue` is queue back-end, while `queue` param is concrette pool. This term is used in around 9 implementations which i checked and in every AMQ documentation. So renaming it to something else === complettely confuse people.